File: 1064-48-1726-tuple-with-same-product/1064-48-1726-tuple-with-same-product.js
var tupleSameProduct = function(nums) {
    let productCount = new Map();
    let count = 0;

    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            let product = nums[i] * nums[j];
            count += (productCount.get(product) || 0) * 8;
            productCount.set(product, (productCount.get(product) || 0) + 1);
        }
    }

    return count;
};

File: 1064-48-1726-tuple-with-same-product/1064-48-1726-tuple-with-same-product.py
from collections import defaultdict

class Solution:
    def tupleSameProduct(self, nums):
        product_count = defaultdict(int)
        count = 0

        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                product = nums[i] * nums[j]
                count += product_count[product] * 8
                product_count[product] += 1

        return count
